components/lcd_touch/esp_lcd_touch/esp_lcd_touch.c

```diff
@@ -256,3 +256,39 @@ esp_err_t esp_lcd_touch_register_interrupt_callback(esp_lcd_touch_handle_t tp, e
 
     return ESP_OK;
 }
+
+esp_err_t esp_lcd_touch_register_interrupt_callback_with_data(esp_lcd_touch_handle_t tp, esp_lcd_touch_interrupt_callback_t callback, void *user_data)
+{
+    esp_err_t ret = ESP_OK;
+    assert(tp != NULL);
+
+    /* Interrupt pin is not selected */
+    if (tp->config.int_gpio_num == GPIO_NUM_NC) {
+        return ESP_ERR_INVALID_ARG;
+    }
+
+    tp->config.interrupt_callback = callback;
+    tp->config.user_data = user_data;
```

Comment:
Am I missing something or this variable is not used anywhere?

Reply:
It can be used in the [ISR callback function](https://github.com/espressif/esp-bsp/blob/162d138b82fb052f33b9b49263974b465b41a6c8/components/lcd_touch/esp_lcd_touch/include/esp_lcd_touch.h#L37) like below:
```c
void lcd_touch_interrupt_callback(esp_lcd_touch_handle_t tp)
{
    void *user_data = tp->config.user_data;
    ....
}
```

The reason why I don't add `user_data` in `typedef void (*esp_lcd_touch_interrupt_callback_t)(esp_lcd_touch_handle_t tp);` is to avoid breaking change.

Reply:
Fair enough, thank you!